File: datasets/Python/Chatgpt/DataStructures/easy4.py
def getMax(operations):
    stack = []
    max_stack = []
    result = []

    for op in operations:
        if op.startswith('1'):
            _, val = op.split()
            val = int(val)
            stack.append(val)
            if not max_stack or val >= max_stack[-1]:
                max_stack.append(val)
        elif op == '2':
            if stack:
                removed = stack.pop()
                if removed == max_stack[-1]:
                    max_stack.pop()
        elif op == '3':
            if max_stack:
                result.append(max_stack[-1])
    return result

# MAIN
n = int(input("Enter number of operations:\n"))
operations = [input() for _ in range(n)]

print("Output:")
for val in getMax(operations):
    print(val)

File: datasets/C++/Chatgpt/DataStructures/easy4.cpp
#include <iostream>
#include <stack>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;
    stack<int> s, maxStack;
    vector<int> results;

    for (int i = 0; i < n; ++i) {
        int type;
        cin >> type;
        if (type == 1) {
            int x;
            cin >> x;
            s.push(x);
            if (maxStack.empty() || x >= maxStack.top()) {
                maxStack.push(x);
            }
        } else if (type == 2) {
            if (!s.empty()) {
                if (s.top() == maxStack.top()) {
                    maxStack.pop();
                }
                s.pop();
            }
        } else if (type == 3) {
            if (!maxStack.empty()) {
                cout << maxStack.top() << endl;
            }
        }
    }

    return 0;
}
